graphula-core/src/Graphula.hs

```diff
@@ -138,9 +139,42 @@ type family GraphulaContext (m :: Type -> Type) (ts :: [Type]) :: Constraint whe
    GraphulaContext m (t ': ts) = (GraphulaNode m t, GraphulaContext m ts)
 
 class EntityKeyGen a where
```

Comment:
Big mood, should `EntityKeyGen` and `HasDependencies` be collapsed into the same typeclass. They are both required for any entity, so keeping them separate is just more boilerplate.

Reply:
This isn't necessary work for this PR.